### DISCUSSION TITLE
Should browse.path contain source folders, or only header folders?

### DISCUSSION THREAD

**Discussion by HampusAdolfsson** [microsoft/vscode-cpptools#11033] [Q&A] [answered]
I'm a developer on an extension that provides intellisense configurations via the TS API, and several users have reported issues with 'Go To Definition'. These users keep their header- and source files in separate folders, and our extension only adds the header folders to `browse.path`. 

The documentation(1) describes `browse.path` only as a list of folders in which to search for include files, but it also seems to be used to determine which source files to index with the tag parser. Should the source folders also be added to `browse.path` or is there something else I've missed here?

1: https://code.visualstudio.com/docs/cpp/c-cpp-properties-schema-reference#_browse-properties
https://code.visualstudio.com/docs/cpp/faq-cpp#_browsepath

**Comment by sean-mcmanus** [comment, marked as answer]
The docs at https://code.visualstudio.com/docs/cpp/c-cpp-properties-schema-reference#_browse-properties are incorrect/misleading.

The browse.path is supposed to include any folders for which you would want to locate a global symbol via symbol search, go to definition, find all references, etc., so it should usually include source files too.
